backend-tests/tests/test_inventory.py

```diff
@@ -529,14 +826,539 @@ def test_search_v2(self, tenants_users):
+                "Failed to save filter, received status code: %d" % rsp.status_code
+            )
+            filter_id = rsp.headers.get("Location").split("/")[-1]
+
+            # Check that we get the exected devices from the set.
+            rsp = invm_v2.with_auth(tenant.api_token).call(
+                "GET", inventory_v2.URL_SAVED_FILTER_SEARCH.format(id=filter_id)
+            )
+            assert rsp.status_code == 200
+
+            devs_recv = sorted([dev["id"] for dev in rsp.json()])
+            devs_exct = sorted(
+                [dev.id for dev in filter(test_case["filter"], tenant.devices)]
+            )
+            assert devs_recv == devs_exct, (
+                "Unexpected device set returned by saved filters, "
+                + "expected: %s, received: %s" % (devs_recv, devs_exct)
+            )
+
+            # Perform perturbations to the device set.
+            # Temporary dict representation mapping device.id -> device
+            devices = {}
+            for dev in tenant.devices:
+                devices[dev.id] = dev
+            # Starting with modifications
+            for fltr, change in test_case["mods"]:
+                for dev in filter(fltr, devices.values()):
+                    for k, v in change.items():
+                        dev.inventory[k] = v
+
+                    rsp = invd.with_auth(dev.token).call(
+                        "PATCH",
+                        inventory.URL_DEVICE_ATTRIBUTES,
+                        body=dict_to_inventoryattrs(dev.inventory),
+                    )
+                    assert rsp.status_code == 200
+                    devices[dev.id] = dev
+
+            # Remove devices
+            for fltr in test_case["remove"]:
+                for dev in filter(fltr, list(devices.values())):
+                    devauthm.with_auth(tenant.api_token).call(
+                        "DELETE", deviceauth.URL_DEVICE.format(id=dev.id)
+                    )
+                    devices.pop(dev.id)
+
+            tenant.devices = list(devices.values())
+
+            # Add new devices
+            add_devices_to_tenant(tenant, test_case["new"])
+
+            # Check that we get the exected devices from the perturbed set.
+            rsp = invm_v2.with_auth(tenant.api_token).call(
+                "GET",
+                inventory_v2.URL_SAVED_FILTER_SEARCH.format(id=filter_id),
+                qs_params={"per_page": len(tenant.devices) + 1},
+            )
+            assert rsp.status_code == 200
+
+            devs_recv = sorted([dev["id"] for dev in rsp.json()])
+            devs_exct = sorted(
+                [dev.id for dev in filter(test_case["filter"], tenant.devices)]
+            )
+            assert devs_recv == devs_exct, (
+                "Unexpected device set returned by saved filters, "
+                + "expected: %s, received: %s" % (devs_recv, devs_exct)
+            )
+
+            mongo_cleanup(mongo)
```

Comment:
Isn't this done automatically by `clean_mongo` fixture?

Reply:
This was just a work-around running mongo_cleanup for each test case. (`clean_mongo` only runs before and after all test cases).

Reply:
Of course. I failed to see that this was per test case.